Sources/GroundState/py_variational_montecarlo.hpp

```diff
@@ -185,8 +200,36 @@ void AddVariationalMonteCarloModule(py::module &m) {
                     to store samples.
                compute_logderivs: Whether to store the logarithmic derivatives
                     of the wavefunction as part of the returned VMC result.
+           Returns:
+               A tuple `(samples, values, gradients)` if `compute_logderivs == True`
+               and a pair `(samples, values)` otherwise. `samples` are visible
+               configurations visited during sampling and `values` are the
+               corresponding values of the logarithm of the wavefunction.
             )EOF");
+#endif
+
+  m_vmc.def("compute_samples", &ComputeSamples, py::arg{"sampler"},
+            py::arg{"nsamples"}, py::arg{"ndiscard"},
+            py::arg{"compute_logderivs"} = true,
+            R"EOF(
+      Computes a sequence of visible configurations based on Monte Carlo sampling using `sampler`.
+
+      Args:
+          sampler: sampler to use for Monte Carlo sweeps.
+          nsamples: number of samples to record (returned number of samples
```

Comment:
may differ how?

Reply:
Yes. Returned number of samples will be `nsamples` rounded up to a multiple of `BatchSize()`.

Reply:
Can this be a bit dangerous when storing the samples in a pre-allocated vector?